kwarchart/src/main/java/com/kwarchart/android/Chart.kt

```diff
@@ -1,52 +1,182 @@
 package com.kwarchart.android
 
 import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.background
 import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.CircleShape
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.graphics.drawscope.DrawScope
 import androidx.compose.ui.graphics.drawscope.scale
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import com.kwarchart.android.enum.LegendPosition
+import com.kwarchart.android.model.Legend
 
 /**
+ * Chart template.
  *
+ * @param modifier Modifier.
+ * @param xAxisName X-axis name.
+ * @param yAxisName Y-axis name.
+ * @param legend Legend to be displayed in the chart.
+ * @param content Chart's children.
  */
 @Composable
 fun Chart(
     modifier: Modifier = Modifier,
-    legendX: String,
-    legendY: String,
-    legendColor: Color = Color.Unspecified,
-    content: @Composable () -> Unit
+    xAxisName: String? = null,
+    yAxisName: String? = null,
+    legend: Legend? = null,
+    content: @Composable ColumnScope.() -> Unit,
 ) {
-    Row(
-        modifier = modifier,
-        verticalAlignment = Alignment.CenterVertically
-    ) {
-        Text(
-            text = legendY,
-            color = legendColor,
-            modifier = Modifier.padding(start = 10.dp)
-        )
-        Column(
-            horizontalAlignment = Alignment.CenterHorizontally
+    Column(modifier = modifier) {
+        if (
+            legend != null &&
+            (legend.position == LegendPosition.TOP_LEFT ||
+            legend.position == LegendPosition.TOP ||
```

Comment:
for fun Chart, pls create a private method pra dli kau taas ang code sa usa ka func and it would be readable
pls check other functions as well, if need pa creatan ug private methods